@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -1200,81 +1205,116 @@ export class TileGeometryCreator {
-                        .projectPoint(new GeoCoordinates(south, east), tmpV)
-                        .toArray(),
-                    ...sourceProjection
-                        .projectPoint(new GeoCoordinates(north, west), tmpV)
-                        .toArray(),
-                    ...sourceProjection
-                        .projectPoint(new GeoCoordinates(north, east), tmpV)
-                        .toArray()
-                ]),
-                3
-            );
-            g.setAttribute("position", posAttr);
-            g.setIndex(new THREE.BufferAttribute(new Uint16Array([0, 1, 2, 2, 1, 3]), 1));
-            const modifier = new SphericalGeometrySubdivisionModifier(
+        function moveTileCenter(geom: THREE.BufferGeometry) {
+            const attr = geom.getAttribute("position") as THREE.BufferAttribute;
+            const posArray = attr.array as Float32Array;
+            for (let i = 0; i < posArray.length; i += 3) {
+                tmpV.set(posArray[i], posArray[i + 1], posArray[i + 2]);
+                projection.reprojectPoint(sourceProjection, tmpV, tmpV);
+                tmpV.sub(tile.center);
+                posArray[i] = tmpV.x;
+                posArray[i + 1] = tmpV.y;
+                posArray[i + 2] = tmpV.z;
+            }
+            attr.needsUpdate = true;
+        }
+
+        // Create plane
+        const { east, west, north, south } = tile.geoBox;
+        const geometry = new THREE.BufferGeometry();
+        const posAttr = new THREE.BufferAttribute(
+            new Float32Array([
+                ...sourceProjection.projectPoint(new GeoCoordinates(south, west), tmpV).toArray(),
+                ...sourceProjection.projectPoint(new GeoCoordinates(south, east), tmpV).toArray(),
+                ...sourceProjection.projectPoint(new GeoCoordinates(north, west), tmpV).toArray(),
+                ...sourceProjection.projectPoint(new GeoCoordinates(north, east), tmpV).toArray()
+            ]),
+            3
+        );
+        geometry.setAttribute("position", posAttr);
+        geometry.setIndex(new THREE.BufferAttribute(new Uint16Array([0, 1, 2, 2, 1, 3]), 1));
+
+        if (createTexCoords) {
+            const uvAttr = new THREE.BufferAttribute(new Float32Array([0, 0, 1, 0, 0, 1, 1, 1]), 2);
+            geometry.setAttribute("uv", uvAttr);
+        }
+
+        if (shouldSubdivide) {
+            const geometries: THREE.BufferGeometry[] = [];
+            const sphericalModifier = new SphericalGeometrySubdivisionModifier(
                 THREE.Math.degToRad(10),
                 sourceProjection
             );
-            modifier.modify(g);
+            const enableMixedLod = mapView.enableMixedLod || mapView.enableMixedLod === undefined;
+
+            if (enableMixedLod) {
+                // Use a [[LodMesh]] to adapt tesselation of tile depending on zoom level
+                for (let zoomLevelOffset = 0; zoomLevelOffset < 4; ++zoomLevelOffset) {
```

Comment:
Do we really need to do this for all zoomlevels and all features?
Especially in zoomed in cases the `LodMesh` will most likely contain 4 times the same geometry.
Especially for buildings this could be quite some memory overhead.

Reply:
Currently it is only used for `BackgroundDatasource` and `WebTileDatasource`. The fix for vector tiles will probably be slightly different and is not part of the current PR.